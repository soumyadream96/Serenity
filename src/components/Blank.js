import React from 'react';
import { Button, withTheme } from 'react-native-paper';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';
import Tree from './Tree';

const Blank = props => {
  const { colors } = props.theme;
  return (
    <View style={[styles.container, { backgroundColor: colors.background }]}>
      <Tree message={props.text} />
      <View style={styles.padding}>
        <Button icon="refresh" mode="text" onPress={props.fetchData}>
          Refresh
        </Button>
      </View>
    </View>
  );
};

export default withTheme(Blank);

Blank.propTypes = {
  theme: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  padding: {
    margin: 4,
  },
});
