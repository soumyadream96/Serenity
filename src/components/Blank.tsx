import React from 'react';
import { Button, withTheme, Theme } from 'react-native-paper';
import { StyleSheet, View } from 'react-native';
import Tree from './Tree';

interface Props {
  theme: Theme;
  text: string;
  fetchData(): void;
}

const Blank = ({ theme, text, fetchData }: Props) => {
  const { colors } = theme;
  return (
    <View style={[styles.container, { backgroundColor: colors.background }]}>
      <Tree message={text} />
      <View style={styles.padding}>
        <Button icon="refresh" mode="text" onPress={fetchData}>
          Refresh
        </Button>
      </View>
    </View>
  );
};

export default withTheme(Blank);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  padding: {
    margin: 4,
  },
});
