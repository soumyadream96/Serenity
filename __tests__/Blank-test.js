import 'react-native';
import React from 'react';
import Blank from '../src/components/Blank';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('In album card renders correctly', () => {
  const onPress = jest.fn();
  renderer.create(
    <Blank fetchData={onPress} />,
  );
});
