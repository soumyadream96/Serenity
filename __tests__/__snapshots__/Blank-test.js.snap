// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blank component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f6f6f6",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "borderRadius": 4,
              "height": 220,
              "width": 209,
            },
            Object {
              "backgroundColor": "#f6f6f6",
            },
          ],
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../src/assets/tree.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Helvetica Neue",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 16,
                "lineHeight": 24,
                "marginVertical": 2,
              },
              undefined,
            ],
            Array [
              Object {
                "fontSize": 16,
                "lineHeight": 24,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      Hello
    </Text>
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 0,
          "minWidth": 64,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "marginLeft": 12,
                "marginRight": -4,
                "width": 16,
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#6200ee",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î—•
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Helvetica Neue",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "HelveticaNeue-Medium",
                  },
                  Object {
                    "fontFamily": "HelveticaNeue-Medium",
                  },
                ],
              ]
            }
          >
            REFRESH
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
